### PR TITLE
TableInteractive - migrate sorting direction to MLv2

### LINKED ISSUE #37527: [MLv2] [FE] VisualizationResult -> TableInteractive -> DimensionInfoPopover
https://github.com/metabase/metabase/blob/965625bbf4efca6285321347543cb3bd8a05199f/frontend/src/metabase/visualizations/components/TableInteractive/TableInteractive.jsx#L676
https://github.com/metabase/metabase/blob/965625bbf4efca6285321347543cb3bd8a05199f/frontend/src/metabase/visualizations/components/TableInteractive/TableInteractive.jsx#L805

### PR DESCRIPTION
**Pull request by ranquild**
Closes https://github.com/metabase/metabase/issues/37527
Depends on https://github.com/metabase/metabase/pull/37648

### CODE REVIEW

File: frontend/src/metabase/visualizations/visualizations/Table.jsx [hunk 1]
```diff
@@ -361,6 +362,7 @@ export default class Table extends Component {
           rows: rows.map(row => columnIndexes.map(i => row[i])),
           results_timezone,
         },
+        question: new Question(card, metadata),
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/visualizations/Table.jsx:365]: Cache question for performance reasons.

File: frontend/src/metabase/visualizations/visualizations/Table.jsx [hunk 2]
```diff
@@ -376,11 +378,31 @@ export default class Table extends Component {
     return getTitleForColumn(cols[columnIndex], series, settings);
   };
 
+  getColumnSortDirection = columnIndex => {
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/visualizations/Table.jsx:381]: Follow the same pattern as with `getColumnTitle`, where the function accepts `columnIndex`

File: frontend/src/metabase/visualizations/visualizations/Table.jsx (hunk 1 above)
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/visualizations/Table.jsx:365]: > Cache question for performance reasons.

That could be an in-code comment.

**Review by kamilmielnik — APPROVED** [review]

**Review by uladzimirdev — APPROVED** [review]

File: frontend/src/metabase/visualizations/visualizations/Table.jsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/visualizations/visualizations/Table.jsx:367]: Done

**Pull request merged**